yoyo/backends.py

```diff
@@ -433,23 +433,32 @@ def to_rollback(self, migrations):
             reversed(topological_sort(ms)), migrations.post_apply
         )
 
-    def apply_migrations(self, migrations, force=False):
+    def apply_migrations(self, migrations, force=False, limit=None):
         if migrations:
-            self.apply_migrations_only(migrations, force=force)
+            self.apply_migrations_only(migrations, force=force, limit=limit)
             self.run_post_apply(migrations, force=force)
 
-    def apply_migrations_only(self, migrations, force=False):
+    def apply_migrations_only(self, migrations, force=False, limit=None):
         """
         Apply the list of migrations, but do not run any post-apply hooks
         present.
         """
         if not migrations:
             return
+        current_count = 0
         for m in migrations:
+            print(m)
+            print(m.applied)
```

Comment:
The issue is here (and in the equivalent rollback function). If I understand it correctly I get _every_ migration in that migrations iterable. That's not a bad thing but with the limits I do something that's non intuitive. If I give it a limit of 1 where I already have a migration applied (applied denoted as `Ma` and unapplied as `Mu`). I'll actually make a no-op instead of apply the next change:

Initial State: `Ma -> Mu -> Mu -> Mu`

Expected Outcome: `Ma -> Ma -> Mu -> Mu`

Actual Outcome Unchanged: `Ma -> Mu -> Mu -> Mu`

In order to do this I believe I need some primitive that tells me if `apply_one` took an action or was a no-op. Then I can:

```
changed_primitive = self.apply_one(m, force=force)
...
if changed_primitive is True:
  current_count += 1
else:
  # Unchanged
```

Reply:
I'm just not sure what the best way to add such a primitive is.